### PR TITLE
Add docs for plugins for beta agent v3.0

### PR DESCRIPTION
**Pull request by lox**
Ok, I got sick of waiting for me to finish the agent docs and am starting to ship them incrementally instead of big bang.

This is the plugins page. I think it will be useful alongside the existing docs.

### CODE REVIEW

File: pages/agent/plugins.md.erb [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+__Plugins are only available in the beta 3.0 version of the agent__
+
+Plugins provide a way to extend the functionality that Buildkite natively offers. Essentially, they are neatly packaged collections of [Hooks](hooks) that run as part of your builds. Buildkite offers a [collection of useful plugins](https://github.com/buildkite-plugins) for things like Docker integration, Github Releases and interaction with Amazon Web Services and Google Cloud. 
+
+<%= toc %>
+
+## Using a plugin
+
+Plugins are only available for [pipelines] expressed in yaml, so all of these examples are going to be using that syntax.
```
Comment (keithpitt) [inline pages/agent/plugins.md.erb]: YAML not yaml :)

File: pages/agent/plugins.md.erb [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+Plugins provide a way to extend the functionality that Buildkite natively offers. Essentially, they are neatly packaged collections of [Hooks](hooks) that run as part of your builds. Buildkite offers a [collection of useful plugins](https://github.com/buildkite-plugins) for things like Docker integration, Github Releases and interaction with Amazon Web Services and Google Cloud. 
+
+<%= toc %>
+
+## Using a plugin
+
+Plugins are only available for [pipelines] expressed in yaml, so all of these examples are going to be using that syntax.  
+
+Plugins are defined per-step and are automatically fetched from github based on the name of the project. For instance, checkout our clown-detection plugin which scans your code for pesky clowns:
```
Comment (keithpitt) [inline pages/agent/plugins.md.erb]: "automatically fetched from where they are hosted (either on GitHub, BitBucket or a private Git repository)"

**Comment by keithpitt** [comment]
Probably needs a section on the syntax of the plugin name?

i.e.

`git@github.com:foo/bar.git#v1.3` for a private plugin

**Comment by keithpitt** [comment]
Also the version is anything that's addressable in Git, so branches, tags, and commits are all valid.

File: pages/agent/plugins.md.erb [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+```yml
+steps:
+  - plugins:
+      detect-clowns:
+        emojis: 🤡
+```
+
+This example will install [buildkite-plugins/detect-clowns-buildkite-plugin](https://github.com/buildkite-plugins/detect-clowns-buildkite-plugin) and run it for this build step.
+
+You could just as easily write your own plugin on GitHub at `myusername/blah-buildkite-plugin` and it would be available as `myusername/blah` in your pipeline.yml.
```
Comment (harrietgrace) [inline pages/agent/plugins.md.erb]: Keep an eye out for using the word 'easy'! We don't want to alienate people who find this difficult 😊

File: pages/agent/plugins.md.erb [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+    command: test.sh
+    parallelism: 25
+    plugins:
+      docker-compose#v1.3.2:
+        run: app
+```
+
+In the 2.x agent, this sort of thing was only possible by either using the limited docker environment variables, or writing a custom script. 
+
+## Anatomy of a Plugin
```
Comment (harrietgrace) [inline pages/agent/plugins.md.erb]: This section seems to be information for people who are writing plugins? Anything globally applicable to using plugins or working with them could be pulled out into the 'using' section or the intro. All the rest could be moved down into the 'writing' section?

**Comment by lox** [comment]
Thanks @harrietgrace and @keithpitt. I've made the changes requested and done a bit of a cleanup. 

@harrietgrace if you have any time, I'd love some assistance with some iterations of this. Very keen to get it out to users :)

File: pages/agent/plugins.md.erb [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+
+Plugins are also able to offer `command` and `environment` hooks, which aren't allowed in per-project hooks. This lets plugins change how commands are executed entirely.
+
+## Security and Reliability
+
+One issue with plugins is that they introduce a third-party dependency in your pipeline. They provide a new way to introduce executable code into your pipeline beyond scripts in your repository and access to the pipeline.yml. 
+
+The practices that we follow are:
+
+ * Always ping to a particular version of the plugin, e.g. `docker-compose#287293c4`. You can refer to any git ref. The agent will check this out the first time it encounters it and then re-use that checked out code.
```
Comment (harrietgrace) [inline pages/agent/plugins.md.erb]: Should this be 'pin' rather than 'ping'?

File: pages/agent/plugins.md.erb [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+
+ * Follow the [Unofficial Bash Strict Mode] and check your work with [shellcheck].
+
+ * Apply special care in plugins to avoid executing arbitrary code passed in by config. As these are just environment variables, anything that can set environment variables can mess with plugin configuration.
+
+For particularly high-security setups, you can disable plugins selectively by removing them from the `BUILDKITE_PLUGINS` variable in your `environment` hook, or preventing them from being loaded entirely with the `no-plugins` [agent configuration](TODO) parameter.
+
+## We love feedback!
+
+Agent plugins are still a new thing and we are evolving them as we learn more. We'd love feedback on how you use them and how we might make them better.
```
Comment (harrietgrace) [inline pages/agent/plugins.md.erb]: Could we remove 'thing' here too please? 😄 

> Agent plugins are still new, and we are evolving them as we learn more

File: pages/agent/plugins.md.erb [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+  # Use the app image built above to run concurrent tests
+  - name: "Docker Test %n"
+    command: test.sh
+    parallelism: 25
+    plugins:
+      docker-compose#v1.3.2:
+        run: app
+```
+
+In the 2.x agent, this sort of thing was only possible by either using the
```
Comment (harrietgrace) [inline pages/agent/plugins.md.erb]: Could we remove the use of 'thing' here? 

> In the 2.x agent, this type of operation was only possible by

or 

> In the 2.x agent, this approach was only possible by

**Comment by lox** [comment]
All instances of "thing" removed @harrietgrace!

File: pages/agent/plugins.md.erb [hunk 8]
```diff
@@ -0,0 +1,166 @@
+
+# Buildkite Agent Plugins
+
+__Plugins are only available in the beta 3.0 version of the agent__
+
+Plugins provide a way to extend the functionality that Buildkite natively offers. 
+Essentially, they are  collections of [Hooks][hooks] that run as part of your
```
Comment (toolmantim) [inline pages/agent/plugins.md.erb]: There's a few of these "Essentially" that might be good to remove

Comment (lox) [reply pages/agent/plugins.md.erb]: Removed a broken link, could only find this one `Essentially`, but I removed it.

**Comment by lox** [comment]
I would love to ship this @toolmantim and @harrietgrace, could you let me know if you want to see any changes?

**Pull request merged**